packages/x-date-pickers/src/DateCalendar/DateCalendar.tsx

```diff
@@ -243,10 +243,10 @@ export const DateCalendar = React.forwardRef(function DateCalendar<TDate>(
   const handleSelectedDayChange = useEventCallback((day: TDate | null) => {
     if (value && day) {
       // If there is a date already selected, then we want to keep its time
-      return setValueAndGoToNextView(mergeDateAndTime(utils, day, value), 'finish');
+      return handleValueChange(mergeDateAndTime(utils, day, value), 'finish');
```

Comment:
There is already a place in this file calling the `setValueAndGoToNextView`, this call was unnecessarily calling the `goToNextView` twice.
In other words: functionality did not seem to change even when `DateCalendar` is used standalone.